lib/bootloader_setup.pm

```diff
@@ -896,11 +896,12 @@ sub tianocore_http_boot {
 sub zkvm_add_disk {
     my ($svirt) = @_;
     if (my $hdd = get_var('HDD_1')) {
-        my $basename = basename($hdd);
-        my $basedir  = svirt_host_basedir();
-        my $hdd_dir  = "$basedir/openqa/share/factory/hdd";
-        my $hdd_path = $svirt->get_cmd_output("find $hdd_dir -name $basename | head -n1 | tr -d '\n'");
-        die "Unable to find image $basename in $hdd_dir" unless $hdd_path;
+        my $basename      = basename($hdd);
+        my $basedir       = svirt_host_basedir();
+        my $hdd_share_dir = "$basedir/openqa/share/factory/hdd";
+        my $hdd_cache_dir = "$basedir/openqa/cache";
+        my $hdd_path      = $svirt->get_cmd_output("find $hdd_share_dir $hdd_cache_dir -name $basename | head -n1 | tr -d '\n'");
```

Comment:
The only problem I have with this, is what if the image is found in both places. Which one will be shown first. 

I'd prefer the worker cache to have priority over the factory dir, given that it goes over nfs...